services/wallet/transfer/transaction_manager_multitransaction.go

```diff
@@ -491,3 +491,67 @@ func (tm *TransactionManager) getVerifiedWalletAccount(address, password string)
 		AccountKey: key,
 	}, nil
 }
+
+func (tm *TransactionManager) removeMultiTransactionByAddress(address common.Address) error {
```

Comment:
Well thought, this works for now. Its complexity also shows that current design of `multi_transaction` table doesn't properly represent its intentions in case of sender and receiver being in `addresses` table. Maybe we can improve on this, bring clarity in the association between the two tables (`multi_transaction` and `transfers`) and simplify this implementation moving forward.

Reply:
> but removing the accounts is not an frequent operation

Actually I was thinking that adding and removing watch only accounts would a common operation, but maybe it is just my testing features mindset :)

>  ... But for any other good reasons and better architecture

For the sake of DB normalization this operation should be one sql query. But it require some effort to align with all the new requirements we gather in the last two years.